src/utils/storage.ts

```diff
@@ -197,15 +200,103 @@ export async function *loadAddressHistory(addresses: string[], storage: IStorage
   }
 }
 
+/**
+ * Get the starting addresses to load from the scanning policy
+ * @param {IStorage} storage The storage instance
+ * @returns {Promise<IScanPolicyLoadAddresses>}
+ */
+export async function scanPolicyStartAddresses(storage: IStorage): Promise<IScanPolicyLoadAddresses> {
+  const scanPolicy = await storage.getScanningPolicy();
+  switch (scanPolicy) {
+    case 'index-limit':
+      const limits = await storage.getIndexLimit();
+      if (!limits) {
+        // This should not happen but it enforces the limits type
+        throw new Error('Index limit is not configured');
+      }
+      return {
+        nextIndex: limits.startIndex,
+        count: limits.endIndex - limits.startIndex + 1,
+      };
+    default:
+    case 'gap-limit':
+      return {
+        nextIndex: 0,
```

Comment:
question: This got me a bit confused: the purpose of this method is to return the _start_ address, right?

So why call this property `nextIndex`? It gives me the impression that this wallet has no transactions on all its mapped addresses ( which would be the interpretation of receiving `0` from `getNextAddress()` )

Should we call this `startIndex` instead?

Reply:
The name is just to fit with the interface and I also think we should use the same types in similar methods.

Reply:
I agree it's better to keep the same types.